libvast/builtins/operators/summarize.cpp

```diff
@@ -741,34 +740,42 @@ class summarize_operator2 : public logical_operator<events, events> {
     // nop
   }
 
-  [[nodiscard]] auto
-  make_physical_operator(const type& input_schema,
-                         operator_control_plane& ctrl) noexcept
-    -> caf::expected<physical_operator<events, events>> override {
-    if (auto aggregation = aggregation::make(input_schema, config_)) {
-      return [aggregation = std::move(*aggregation), &ctrl](
-               generator<table_slice> input) mutable -> generator<table_slice> {
-        for (auto&& slice : input) {
-          aggregation.add(to_record_batch(slice));
-          co_yield {};
-        }
-        if (auto batch = aggregation.finish()) {
+  auto initialize(const type& schema) const
+    -> caf::expected<state_type> override {
+    auto result = aggregation::make(schema, config_);
+    if (!result) {
+      VAST_WARN("summarize operator does not apply to schema {} and discards "
+                "events: {}",
+                schema, result.error());
+      return std::nullopt;
+    }
+    return *result;
+  }
+
+  auto process(table_slice slice, state_type& state) const
+    -> output_type override {
+    if (state) {
+      state->add(to_record_batch(slice));
+    }
+    return {};
+  }
+
+  auto finish(std::unordered_map<type, state_type> states,
+              operator_control_plane& ctrl) const
```

Comment:
Besides `summarize`, there currently is no need for this function. A schema-specific `sort` or some `taste-tail` could use this. But I don't think we want those, so its fine by me if we remove it.

Reply:
Actually, let's perhaps do this as part of the `summarize` rewrite, because we could also change the API to make `summarize` possible. No need to rewrite this before potentially rewriting it again.